pytest_pyvista.py

```diff
@@ -1,21 +1,168 @@
 # -*- coding: utf-8 -*-
 
+import os
+import pathlib
+import platform
+import warnings
+
 import pytest
+import pyvista
+from pyvista._vtk import VTK9
 
 
 def pytest_addoption(parser):
-    group = parser.getgroup('pyvista')
+    """Adds new flag options to the pyvista plugin."""
+
+    group = parser.getgroup("pyvista")
+    group.addoption(
+        "--reset_image_cache",
+        action="store_true",
+        help="Reset the images in the PyVista cache.",
+    )
     group.addoption(
-        '--foo',
-        action='store',
-        dest='dest_foo',
-        default='2022',
-        help='Set the value for the fixture "bar".'
+        "--ignore_image_cache", action="store_true", help="Ignores the image cache."
     )
+    group.addoption(
+        "--fail_extra_image_cache",
+        action="store_true",
+        help="Enables failure if image cache does not exist.",
+    )
+    parser.addini("HELLO", "Dummy pytest.ini setting")
+
+
+class VerifyImageCache:
+    """Control image caching for testing.
+
+    Image cache files are names according to ``test_name``.
+    Multiple calls to an instance of this class will append
+    `_X` to the name after the first one.  That is, files
+    ``{test_name}``, ``{test_name}_1``, and ``{test_name}_2``
+    will be saved if called 3 times.
+
+    Parameters
+    ----------
+    test_name : str
+        Name of test to save.  Sets name of image cache file.
+
+    """
+
+    reset_image_cache = False
+    ignore_image_cache = False
+    fail_extra_image_cache = False
+
+    high_variance_tests = False
+    windows_skip_image_cache = False
+    macos_skip_image_cache = False
+
+    def __init__(
+        self,
+        test_name,
+        *,
+        cache_dir=None,
+        error_value=500,
+        warning_value=200,
+        var_error_value=1000,
+        var_warning_value=1000,
+    ):
+        self.test_name = test_name
+
+        if cache_dir is None:
+            # Reset image cache with new images
+            this_path = pathlib.Path(__file__).parent.absolute()
```

Comment:
Will this path end up in a site-packages? If so, maybe it should not have a default.

Reply:
I'd recommend using `appdirs` for this

Reply:
This dir is the cached images, so this would be in the repo somewhere likely.  We _could_ allow for the generated images to be stored in a configurable location so that developers could check differences more easily.  I think `appdirs` would be a good default for this second usage.

Reply:
I added the option `--image_cache_dir` to the plugin, so the user can select where the images are saved. 

I tried to also implement `appdirs` as the default. It works fine with `PyVista` tests, but the plugin tests fail with this change because it can't access the `CSIDL_LOCAL_APPDATA` variable from the windows system for some reason.